hail/python/hail/ir/ir.py

```diff
@@ -1887,6 +1887,56 @@ def renderable_uses_scan_context(self, i: int):
         return i == 1
 
 
+class AggFold(IR):
+    @typecheck_method(zero=IR, seq_op=IR, comb_op=IR, accum_name=str, other_accum_name=str, is_scan=bool)
+    def __init__(self, zero, seq_op, comb_op, accum_name, other_accum_name, is_scan):
+        super().__init__(zero, seq_op, comb_op)
+        self.zero = zero
+        self.seq_op = seq_op
+        self.comb_op = comb_op
+        self.accum_name = accum_name
+        self.other_accum_name = other_accum_name
+        self.is_scan = is_scan
+
+        if self.comb_op.free_vars != set([accum_name, other_accum_name]):
```

Comment:
I think this is unnecessarily strict. It should be possible to *not* use one of the vars. I think you want set subtraction == empty set.

Reply:
Great point, thanks